lib/Dialect/SV/Transforms/PrettifyVerilog.cpp

```diff
@@ -52,6 +55,40 @@ static bool isVerilogUnaryOperator(Operation *op) {
   return false;
 }
 
+/// Sink an operation into the same block where it is used.  This will clone the
+/// operation so it can be sunk into multiple blocks. If there are no more uses
+/// in the current block, the op will be removed.
+void PrettifyVerilogPass::sinkOpToUses(Operation *op) {
+  assert(mlir::MemoryEffectOpInterface::hasNoEffect(op) &&
+         "Op with side effects cannot be sunk to its uses.");
+  auto block = op->getBlock();
+  // This maps a block to the block local instance of the op.
+  SmallDenseMap<Block *, Value, 8> blockLocalValues;
+  for (auto &use : llvm::make_early_inc_range(op->getUses())) {
+    // If the current use is in the same block as the operation, there is
+    // nothing to do.
+    auto localBlock = use.getOwner()->getBlock();
+    if (block == localBlock)
+      continue;
+    // Find the block local clone of the operation. If there is not one already,
+    // the op will be cloned in to the block.
+    auto localValue = blockLocalValues.lookup(localBlock);
+    if (!localValue) {
```

Comment:
Further nit, but since Value is nullable you can do:

```
auto &localValue = blockLocalValues[localBlock];
if (!localValue) {
   localValue = <materialize it>
}

use.set(localValue)
```

To avoid to map lookups.

Reply:
Thanks! I wasn't sure how to do this